tls-attestation/draft-fossati-tls-attestation.md

```diff
@@ -486,7 +493,98 @@ Auth | {CertificateVerify*}
    indicates the attestation type carried in the Certificate payload.
    Note that only a single value is permitted in the 
    server_attestation_type extension when carried in the EncryptedExtensions
-   message. 
+   message.
+
+# TPM Attestation
+
+One type of hardware RoT that can be used to prove control over a certain platform is the {{TPM2.0}}. TPMs natively offer interfaces for producing both key and platform attestation tokens.
+
+## Platform Attestation
+
+Platform Configuration Registers (PCRs) represent the core mechanism in TPMs for measuring and conveying information about the platform state via remote attestation. While specifications exist for assigning individual PCRs to specific software components, the choice of which combination of PCRs to include for any attestation procedure (and which hashing algorithm to use) is left to the parties involved. The agreement over and configuration of the PCR selection falls outside the scope of this specification and thus expected to occur out-of-band.
+
+The attestation evidence is produced through the TPM2_Quote operation. The evidence along with all other relevant metadata is transmitted in a format derived from the {{WebAuthn}} Attestation Statements. This format and the workflows around it are defined below.
+
+### TPM Platform Attestation Statement Format
+
+The TPM Platform Attestation Statement is a modified version of the TPM Attestation Statement Format which covers key attestation tokens.
+
+~~~~
+    tpmPlatStmtFormat = {
+                          ver: "2.0",
+                          (
+                              alg: COSEAlgorithmIdentifier,
+                              x5c: [ pakCert: bytes, * (caCert: bytes) ]
+                          )
+                          sig: bytes,
+                          attestInfo: bytes,
+                        }
+~~~~
+{: #figure-tpm-plat-attest-stmt title="TPM Platform Attestation Statement Format"}
+
+- _ver_: The version of the TPM specification to which the signature conforms.
+- _alg_: A COSEAlgorithmIdentifier containing the identifier of the algorithm used to generate the attestation signature.
+- _x5c_: A certificate for the PAK, followed by its certificate chain (in X.509 encoding).
+  - _pakCert_: The PAK certificate used for the attestation, in X.509 encoding.
```

Comment:
There are different ways to encode a single X.509 certificate as well, including a PEM encoding or a DER encoding.